cmake/FindMASA.cmake

```diff
@@ -0,0 +1,47 @@
+# Find MASA library
+#
+# Set MASA_DIR to the base directory where the package is installed
+#
+# Sets 3 variables
+#   - MASA_INCLUDE_DIRS
+#   - MASA_LIBRARY
+#   - MASA_VERSION
+#
+
+find_package(PkgConfig QUIET)
+pkg_check_modules(PC_MASA QUIET MASA)
+
+find_path(MASA_INCLUDE_DIRS_TEMP
+    NAMES masa.h
+    PATHS ${PC_MASA_INCLUDE_DIRS}
+    HINTS ${MASA_DIR} ${CMAKE_INSTALL_PREFIX}
+    PATH_SUFFIXES include
+)
+
+find_library(MASA_LIBRARY_TEMP
+  NAMES masa
+  HINTS ${MASA_DIR} ${CMAKE_INSTALL_PREFIX}
+  PATH_SUFFIXES lib)
+
+set(MASA_VERSION ${PC_MASA_VERSION})
+
+mark_as_advanced(MASA_FOUND MASA_INCLUDE_DIRS MASA_LIBRARY_TEMP MASA_VERSION)
+
+include(FindPackageHandleStandardArgs)
+find_package_handle_standard_args(MASA
+    REQUIRED_VARS MASA_INCLUDE_DIRS_TEMP MASA_LIBRARY_TEMP
+    VERSION_VAR MASA_VERSION
+)
+
+if(MASA_FOUND)
+    set(MASA_INCLUDE_DIRS ${MASA_INCLUDE_DIRS_TEMP})
+    set(MASA_LIBRARY ${MASA_LIBRARY_TEMP})
```

Comment:
I am curious. What is the reason for this indirection? i.e., from TEMP variable to the actual variable?

Reply:
In the example I got it from, it looks like it uses pkgconfig to populate temporary variables for headers and libraries and if they are found, then it sets the actual variables to these values. I assumed he had a reason for it, but I'm not sure why he had this intermediate step. I guess it doesn't really look necessary.